dex/networks/eth/txdata.go

```diff
@@ -43,22 +43,22 @@ func ParseInitiateData(calldata []byte) (*common.Address, [32]byte, int64, error
 	if len(args) != numInputArgs {
 		return fail(fmt.Errorf("expected %v input args but got %v", numInputArgs, len(args)))
 	}
-	locktime, ok := args[0].value.(*big.Int)
+	initiations, ok := args[0].value.([]struct {
+		RefundTimestamp *big.Int       `json:"refundTimestamp"`
+		SecretHash      [32]byte       `json:"secretHash"`
+		Participant     common.Address `json:"participant"`
+		Value           *big.Int       `json:"value"`
+	})
```

Comment:
Looks like we need the json tags to convert here? I guess it's fine because this won't build if you can't cast to `ETHSwapInitiation` below. I would have thought structs have default tags like this but I guess not.

Reply:
Yeah the fakeArg in the code you copy pasted from geth adds the tags and you need the correct tags to do the type conversion.
Someone was trying to update go to ignore tags at one point, but I guess it didn't get implemented:
https://go.googlesource.com/proposal/+/master/design/16085-conversions-ignore-tags.md